automation/run/run.go

```diff
@@ -2,19 +2,40 @@
 package run
 
 import (
+	"encoding/json"
 	"fmt"
 	"io"
+	"sort"
 	"time"
 
 	"github.com/google/uuid"
+	"github.com/qri-io/qri/automation/workflow"
 	"github.com/qri-io/qri/event"
 )
 
+var (
+	// ErrNoID indicates the run.State has no run ID
+	ErrNoID = fmt.Errorf("no run ID")
+	// ErrNoWorkflowID indicates the run.State has no workflow.ID
+	ErrNoWorkflowID = fmt.Errorf("no workflow ID")
+)
+
 // NewID creates a run identifier
 func NewID() string {
 	return uuid.New().String()
 }
 
+// // ID is a run identifier
+// type ID string
+
+// // NewID creates a run identifier
+// func NewID() ID {
+// 	return ID(uuid.New().String())
+// }
+
+// // String returns the ID as the underlying string
+// func (id ID) String() string { return string(id) }
+
```

Comment:
I'm weirdly attached to the idea of having a separate run id type. Maybe because I assume that it'll be easy to switch the run id with some other identifying string?